app/common_handlers.py

```diff
@@ -476,9 +478,16 @@ async def post(self, request):
                     session['attributes']['case_type'] = uprn_return['caseType']
                     session['attributes']['address_level'] = uprn_return['addressLevel']
                     session.changed()
-                    raise HTTPFound(
-                        request.app.router['RequestCodeEnterMobile:get'].url_for(request_type=sub_user_journey,
-                                                                                 display_region=display_region))
+
+                    if uprn_return['caseType'] == 'CE' and uprn_return['addressLevel'] == 'E':
```

Comment:
This method is getting quite long and nested. Consider future improvement to decompose this method.